### PR TITLE
Red cologne

### PR DESCRIPTION
**Pull request by aifodu**
Please review.

### What happened

- Fully migrated the main datasource from mutex to postgres
- Restructured the services module and re-wrote parts of the core functionalities 
- Worked on the docker-compose.yml to create a database and table when running `docker compose up`

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      test: pg_isready -U postgres -h 127.0.0.1
+
+  pgadmin:
+    image: dpage/pgadmin4
+    container_name: pgadmin
+    environment:
+      - PGADMIN_DEFAULT_EMAIL=mail@villagesquare.com
+      - PGADMIN_DEFAULT_PASSWORD=password
+    ports:
+      - '5555:80'
```
Comment (thatkazuk1) [inline docker-compose.yml:28]: Why are we using this — `pgadmin`?

**Review by thatkazuk1 — CHANGES_REQUESTED** [review]
Is `pgadmin` needed?

File: docker-compose.yml (hunk 1 above)
Comment (aifodu) [reply docker-compose.yml:28]: There is no serious reason to have it there. It was merely to have `pgadmin `running in order to view that database. Thinking about it some more, we really don't need it.

**Comment by aifodu** [comment]
A future pull request will take care of this.

**Pull request closed without merging**